plugins/colorschemes/modus.nix

```
@@ -0,0 +1,95 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "modus";
+  isColorscheme = true;
+  originalName = "modus-themes-nvim";
+  defaultPackage = pkgs.vimPlugins.modus-themes-nvim;
+
+  maintainers = [ helpers.maintainers.nwjsmith ];
+
+  settingsOptions = {
+    style =
+      helpers.defaultNullOpts.mkEnumFirstDefault
+        [
+          "modus_operandi"
+          "modus_vivendi"
+        ]
+        ''
+          The theme comes in two styles, `modus_operandi`, a dark variant `modus_vivendi`.
+        '';
+
+    variant =
+      helpers.defaultNullOpts.mkEnumFirstDefault
+        [
+          "default"
+          "tinted"
+          "deuteranopia"
+          "tritanopia"
+        ]
+        ''
+          The theme comes in four variants `default`, `tinted`, `deuteranopia`, and `tritanopia`.
+        '';
+
+    transparent = helpers.defaultNullOpts.mkBool false ''
+      Disable setting the background color.
+    '';
+
+    dim_inactive = helpers.defaultNullOpts.mkBool false ''
+      Dims inactive windows.
+    '';
+
+    styles = {
+      comments = helpers.defaultNullOpts.mkHighlight "{italic = true;}" "" ''
+        Define comments highlight properties.
+      '';
+
+      keywords = helpers.defaultNullOpts.mkHighlight "{italic = true;}" "" ''
+        Define keywords highlight properties.
+      '';
+
+      functions = helpers.defaultNullOpts.mkHighlight "{}" "" ''
+        Define functions highlight properties.
+      '';
+
+      variables = helpers.defaultNullOpts.mkHighlight "{}" "" ''
+        Define variables highlight properties.
+      '';
+    };
+
+    on_colors = helpers.defaultNullOpts.mkLuaFn "function(colors) end" ''
+      Override specific color groups to use other groups or a hex color.
+      Function will be called with a `ColorScheme` table.
+      `@param colors ColorScheme`
```

```suggestion
      Override specific color groups to use other groups or a hex color.
      Function will be called with a `ColorScheme` table.

      ```
        @param colors ColorScheme
      ```
```